File: framework/src/main/java/com/software/dm/swallow/stormy/security/match/bdia/bean/T_theme_url_rule.java
package com.software.dm.swallow.stormy.security.match.bdia.bean;

import java.util.List;

/**
 * 
 * @Description
 * @author DM
 * @date 2017
 * @version v1.0.0.1
 * 
 */
public final class T_theme_url_rule {
	
	public static final int ACTION_THEME = 1;
	
	
	private long n_id;
	private String domain_comp;
	private String rule;
	private int rule_type;
	private List<T_theme_basic_type_rel> themeTypes;

	public T_theme_url_rule(long n_id, String domain_comp, String rule, int rule_type, List<T_theme_basic_type_rel> themeTypes) {
		this.n_id = n_id;
		this.domain_comp = domain_comp;
		this.rule = rule;
		this.rule_type = rule_type;
		this.themeTypes = themeTypes;
	}

	public long getN_id() {
		return n_id;
	}

	public String getDomain_comp() {
		return domain_comp;
	}

	public String getRule() {
		return rule;
	}

	public int getRule_type() {
		return rule_type;
	}

	public List<T_theme_basic_type_rel> getThemeTypes() {
		return themeTypes;
	}

}

File: framework/src/main/scala/com/software/dm/swallow/stormy/scala/security/addrlib/bdia/bean/T_theme_url_rule.scala
package com.software.dm.swallow.stormy.scala.security.addrlib.bdia.bean

import scala.collection.mutable.ListBuffer


class T_theme_url_rule {
  val ACTION_THEME = 1


  private var n_id = 0L
  private var domain_comp: String  = null
  private var rule : String = null
  private var rule_type = 0
  private var themeTypes : ListBuffer[T_theme_basic_type_rel]  = null

  def this(n_id: Long, domain_comp: String, rule: String, rule_type: Int, themeTypes:  ListBuffer[T_theme_basic_type_rel]) {
    this()
    this.n_id = n_id
    this.domain_comp = domain_comp
    this.rule = rule
    this.rule_type = rule_type
    this.themeTypes = themeTypes
  }

  def getN_id: Long = n_id

  def getDomain_comp: String = domain_comp

  def getRule: String = rule

  def getRule_type: Int = rule_type

  def getThemeTypes:  ListBuffer[T_theme_basic_type_rel] = themeTypes
}
